### DISCUSSION TITLE
OnNavigatedTo and OnNavigatedFrom not called in Prism when switching tabs via tab navigation

### DISCUSSION THREAD

**Discussion by Frizbi22** [PrismLibrary/Prism#3283] [.NET MAUI]
I am testing Prism with a Tabbed Pages navigation flow in the Prism sample app, and I have observed the following behavior:

1. When I switch between tabs using the tab bar navigation (e.g., selecting Tab B from Tab A), the OnNavigatedTo method in Tab B's ViewModel and the OnNavigatedFrom method in Tab A's ViewModel are not being called.

2. However, when I use a button within the page to navigate programmatically to another tab using INavigationService.NavigateAsync, both lifecycle methods (OnNavigatedTo and OnNavigatedFrom) are triggered as expected.

Could someone please explain why this happens? Is this the intended behavior for Prism's navigation lifecycle with Tabbed Pages, or is there a recommended way to handle OnNavigatedTo and OnNavigatedFrom when switching tabs via the tab bar?

https://github.com/user-attachments/assets/fd9d0bc9-495c-44c0-b0f9-adc23cbc15ff

**Comment by Frizbi22** [comment]
Or is this the right way to go ? 

using MauiModule.ViewModels;
using Prism.Mvvm;
using Prism.Navigation;
using System;

public class ViewCViewModel : ViewModelBase, IActiveAware, IPageLifecycleAware
{
    private bool _isActive;

    // Property to track whether the ViewModel is active
    public bool IsActive
    {
        get => _isActive;
        set
        {
            if (_isActive != value)
            {
                _isActive = value;
                OnIsActiveChanged(); // Trigger the event when IsActive changes
            }
        }
    }

    // Event triggered when IsActive changes
    public event EventHandler IsActiveChanged;

    // Constructor
    public ViewCViewModel(BaseServices baseServices)
        : base(baseServices)
    {
        // Optionally subscribe to the IsActiveChanged event
        IsActiveChanged += OnIsActiveChangedHandler;
    }

    // Called when navigation to this ViewModel occurs
    public override void OnNavigatedTo(INavigationParameters parameters)
    {
        base.OnNavigatedTo(parameters);

        // Example logic for when navigation arrives at this ViewModel
        if (IsActive)
        {
            // Perform actions specific to being active
            Console.WriteLine("ViewC is now active.");
        }
    }
    public void OnAppearing()
    {
        Messages.Add("View Appearing");
    }

    public void OnDisappearing()
    {
        Messages.Add("View Disappearing");
    }
    // Called when navigation away from this ViewModel occurs
    public override void OnNavigatedFrom(INavigationParameters parameters)
    {
        base.OnNavigatedFrom(parameters);

        // Example logic for when navigation leaves this ViewModel
        Console.WriteLine("Navigated away from ViewC.");
    }

    // Determines if the navigation request is targeting this ViewModel
    public bool IsNavigationTarget(NavigationContext navigationContext)
    {
        return true; // This ViewModel is always a valid target
    }

    // Helper method to raise the IsActiveChanged event
    protected virtual void OnIsActiveChanged()
    {
        IsActiveChanged?.Invoke(this, EventArgs.Empty);
    }

    // Example handler for the IsActiveChanged event
    private void OnIsActiveChangedHandler(object sender, EventArgs e)
    {
        if (IsActive)
        {
            // Logic for when the ViewModel becomes active
            Console.WriteLine("ViewC is active.");
        }
        else
        {
            // Logic for when the ViewModel becomes inactive
            Console.WriteLine("ViewC is inactive.");
        }
    }
}

**Reply by trobert-hydromat** [reply]
Did you ever find an answer? I'm having a similar issue. @Frizbi22
